src/sonic-yang-models/yang-models/sonic-syslog.yang

```diff
@@ -9,22 +9,64 @@ module sonic-syslog {
         prefix inet;
     }
 
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "Syslog YANG Module for SONiC OS: remote syslog logging";
+
+    revision 2022-04-18 {
+        description "Initial revision.";
+    }
+
+    typedef vrf-device {
+        description "Represents syslog VRF device";
+        type enumeration {
+            enum default;
+            enum mgmt;
+        }
+    }
+
     container sonic-syslog {
 
         container SYSLOG_SERVER {
 
-            description "SYSLOG SERVER part of config_db.json";
+            description "SYSLOG_SERVER part of config_db.json";
 
             list SYSLOG_SERVER_LIST {
 
                 key "server_address";
 
                 leaf server_address {
-                    description "syslog server ip address";
+                    description "Syslog server IP address";
                     type inet:ip-address;
                 }
-            } /* end of list SYSLOG_SERVER_LIST */ 
 
-        } /* end of container SYSLOG_SERVER */
+                leaf source {
```

Comment:
Should the source ip family type and server_address ip family type match?

Reply:
@dgsudharsan for sure. The CLI has this kind of check

Reply:
Then we need to add this in yang as well

Reply:
@dgsudharsan done